cli/cmd/proxy/read/filters_test.go

```diff
@@ -0,0 +1,448 @@
+				},
+			},
+		},
+		"Filter combo": {
+			address: "235",
+			port:    20000,
+			expected: []Endpoint{
+				{
+					Address: "192.168.47.235:20000",
+					Weight:  1,
+					Status:  "HEALTHY",
+				},
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			actual := FilterEndpoints(given, tc.address, tc.port)
+			require.Equal(t, tc.expected, actual)
+		})
+	}
+}
+
+func TestFilterListeners(t *testing.T) {
+	given := []Listener{
+		{
+			Name:    "public_listener",
+			Address: "192.168.69.179:20000",
+			FilterChain: []FilterChain{
+				{
+					FilterChainMatch: "Any",
+					Filters:          []string{"* -> local_app/"},
+				},
+			},
+			Direction:   "INBOUND",
+			LastUpdated: "2022-06-09T00:39:27.668Z",
+		},
+		{
+			Name:    "outbound_listener",
+			Address: "127.0.0.1:15001",
+			FilterChain: []FilterChain{
+				{
+					FilterChainMatch: "10.100.134.173/32, 240.0.0.3/32",
+					Filters:          []string{"-> client.default.dc1.internal.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00.consul"},
+				},
+			},
+			Direction:   "OUTBOUND",
+			LastUpdated: "2022-05-24T17:41:59.079Z",
+		},
+	}
+
+	cases := map[string]struct {
+		address  string
+		port     int
+		expected []Listener
+	}{
+		"No filter": {
+			address: "",
+			port:    -1,
+			expected: []Listener{
+				{
+					Name:    "public_listener",
+					Address: "192.168.69.179:20000",
+					FilterChain: []FilterChain{
+						{
+							FilterChainMatch: "Any",
+							Filters:          []string{"* -> local_app/"},
+						},
+					},
+					Direction:   "INBOUND",
+					LastUpdated: "2022-06-09T00:39:27.668Z",
+				},
+				{
+					Name:    "outbound_listener",
+					Address: "127.0.0.1:15001",
+					FilterChain: []FilterChain{
+						{
+							FilterChainMatch: "10.100.134.173/32, 240.0.0.3/32",
+							Filters:          []string{"-> client.default.dc1.internal.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00.consul"},
+						},
+					},
+					Direction:   "OUTBOUND",
+					LastUpdated: "2022-05-24T17:41:59.079Z",
+				},
+			},
+		},
+		"Filter address 127.0.0.1": {
+			address: "127.0.0.1",
+			port:    -1,
+			expected: []Listener{
+				{
+					Name:    "outbound_listener",
+					Address: "127.0.0.1:15001",
+					FilterChain: []FilterChain{
+						{
+							FilterChainMatch: "10.100.134.173/32, 240.0.0.3/32",
+							Filters:          []string{"-> client.default.dc1.internal.bc3815c2-1a0f-f3ff-a2e9-20d791f08d00.consul"},
+						},
+					},
+					Direction:   "OUTBOUND",
+					LastUpdated: "2022-05-24T17:41:59.079Z",
+				},
+			},
+		},
+		"Filter port 20000": {
+			address: "",
+			port:    20000,
+			expected: []Listener{
+				{
+					Name:    "public_listener",
+					Address: "192.168.69.179:20000",
+					FilterChain: []FilterChain{
+						{
+							FilterChainMatch: "Any",
+							Filters:          []string{"* -> local_app/"},
+						},
+					},
+					Direction:   "INBOUND",
+					LastUpdated: "2022-06-09T00:39:27.668Z",
+				},
+			},
+		},
+	}
+
```

Comment:
Are you missing, address and port combo?

Reply:
Yes! Added